src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -148,38 +159,26 @@
     @Override
     public void beginTree(DetailAST rootAST) {
         classes = new ArrayDeque<>();
+        innerClasses = new TreeMap<>();
```

Comment:
Using `TreeMap` here as it does not allow to use `null` key values (pitest issue),